crypto/s2n_aead_cipher_aes_gcm.c

```diff
@@ -23,26 +23,126 @@
 #include "utils/s2n_safety.h"
 #include "utils/s2n_blob.h"
 
+#if defined(OPENSSL_IS_BORINGSSL) || defined(OPENSSL_IS_AWSLC)
+#define S2N_AEAD_AES_GCM_AVAILABLE_BSSL_AWSLC
```

Comment:
Do we need "\_BSSL\_AWSLC\_" here? Can we just do "S2N_AEAD_AES_GCM_AVAILABLE" instead? I worry that if we add more libcryptos that use this, the define just gets longer.

Reply:
I used that naming scheme as a way to delineate implementations into corresponding libcryptos. It makes it obvious from reading the code that e.g. "okay, this implementation is for AWS-LC". I found that the tradeoff with a longer macro name was worth it for the added clarity. Happy to change if you find it is not.

On a side not: what is the intended behaviour for the "availability check"? Let's say that the symbol `EVP_aes_256_gcm` is actually missing, the function `s2n_aead_cipher_aes256_gcm_available` returns 0. But the code wouldn't compile because the symbol is missing and there is no pre-compilation removing it. The if/def pattern I used is the original pattern used for AES-GCM. Maybe we should use the if/def pattern I used for chacha20-poly1305, where all functions fail if the cipher is not supported?

Reply:
We might want another opinion since this is subjective, but I'd say I'm against that tradeoff. I think the #if itself makes it clear what libcrypto implementations support this define/mode, and the define/mode should just describe what it actually toggles.

And for the side note: I'm not sure, I haven't looked deeply into that OpenSSL code. Is it possible we previously assumed EVP_aes_256_gcm will always exist, but might sometimes return false and be unusable anyway?

Reply:
Shortened the macro name.